pkg/controllermanager/controller/seed/seed_extension_check_control.go

```
@@ -95,17 +96,25 @@ func (r *extensionCheckReconciler) Reconcile(ctx context.Context, request reconc
 		}
 		return reconcile.Result{}, fmt.Errorf("error retrieving object from store: %w", err)
 	}
-	if err := r.reconcile(ctx, seed); err != nil {
-		return reconcile.Result{}, err
-	}
 
+	r.reconcile(ctx, log, seed)
 	return reconcile.Result{RequeueAfter: r.config.SyncPeriod.Duration}, nil
 }
 
-func (r *extensionCheckReconciler) reconcile(ctx context.Context, seed *gardencorev1beta1.Seed) error {
+func (r *extensionCheckReconciler) reconcile(ctx context.Context, log logr.Logger, seed *gardencorev1beta1.Seed) {
+	condition := helper.GetOrInitCondition(seed.Status.Conditions, gardencorev1beta1.SeedExtensionsReady)
+	extensionsReadyThreshold := getThresholdForCondition(r.config.ConditionThresholds, gardencorev1beta1.SeedExtensionsReady)
+
 	controllerInstallationList := &gardencorev1beta1.ControllerInstallationList{}
 	if err := r.gardenClient.List(ctx, controllerInstallationList, client.MatchingFields{core.SeedRefName: seed.Name}); err != nil {
-		return err
+		log.Error(err, "Could not fetch ControllerInstallations")
```

Can you explain, why errors are simply ignored now?
Shouldn't we retry on errors? 👀

---

We do similar things for the seed care controller: https://github.com/gardener/gardener/blob/c628f5cf013a1a3242e170872908ace39062fe34/pkg/gardenlet/controller/seed/seed_care_control.go#L105-L114
Although, I was initially wondering whether the `List` call here should be handled the same way.

---

We should rather limit the exponential backoff to the sync period IMHO

---

👍 the exponential backoff limit could be handled as part of #6636 right?

---

Yes, I can do it as part of this PR